### ISSUE TITLE
Bash always seems to run git when opening a terminal in a vscode container

### ISSUE THREAD

**Issue by Jademalo** [microsoft/vscode-dev-containers#1381]
Issue Type: <b>Bug</b>

I created a new Ubuntu dev container using "Add development container files". After starting the container, I attempted to start a new bash terminal through Terminal->New Terminal.

However, when the terminal loads, there is no output and the process in the list to the right hand side says git.
If I Ctrl+C on the terminal window, git closes and I get back to bash. however, as soon as I enter another command, it immediately starts running git again.

I've got no idea what is causing this, and my container currently has no special configuration. it is simply a basic ubuntu container built with the automatic config within VS Code.

Extension version: 0.224.2
VS Code version: Code 1.65.2 (c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1, 2022-03-10T14:33:55.248Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No
Remote OS version: Linux x64 5.10.16.3-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8086K CPU @ 4.00GHz (12 x 4008)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.84GB (15.98GB free)|
|Process Argv|--crash-reporter-id 75d2bff3-b4ef-4cb2-9b19-e7eeeba2b392|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container: Ubuntu|
|OS|Linux x64 5.10.16.3-microsoft-standard-WSL2|
|CPUs|Intel(R) Core(TM) i7-8086K CPU @ 4.00GHz (12 x 4008)|
|Memory (System)|15.55GB (12.60GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vsclayoutctrt:30451275
dsvsc009:30452663
pythonvspyt640:30450904
vscscmwlcmc:30438804
vscgsvid2:30447481
pynewfile477:30450038

```

</details>

**Comment by Chuxel** [comment]
You can disable the prompt using `git config codespaces-theme.hide-status 1`.  If you have a very larger repo this can happen.

See https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/common.md#speeding-up-the-command-prompt-in-large-repositories

**Comment by Chuxel** [comment]
Yeah NTFS is going to be noticeably slower that if you use the WSL filesystem (or a volume, which is a Linux filesystem). It's effectively going through a share. NTFS is also less efficient with lots of small files which adds to the problem.

This is a tip on reusing the git credential manager on the Windows side in WSL if you are not ware of that: https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-git#git-credential-manager-setup

Windows and Linux also have different line ending defaults, so generally I'd recommend git cloning in WSL and using Remote - WSL extension to edit files directly if you need to rather than flipping back to Windows.  VS Code will keep things with the line endings that were originally used either way, but it helps with new files.

**Comment by Jademalo** [comment]
Alright, that makes a lot of sense, thanks.

Biggest issue personally is I'm using a piece of windows software that edits map data for a game, and it entirely fails editing when the code is hosted on WSL due to a Qt bug. Seems I'm stuck between a rock and a hard place.

Ah well, at least I know what the issues are now, and how to make the terminal usable at least. Appreciate the help, ty!

**Comment by farzonl** [comment]
Is there a work around for this issue. I've tried adding:
```
.git
.git/
**/.git
```
to a `.dockerignore` file but remote container is not respecting the `.dockerignore` preferences. As a result I'm stuck with git bash when I don't need it.

**Comment by artm** [comment]
> This is a tip on reusing the git credential manager on the Windows side in WSL if you are not ware of that: https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-git#git-credential-manager-setup

this tip breaks git support inside the dev containers. It suggests to configure a windows executable as credential helper, but the executable isn't available inside the container and git refuses to start. Bind-mounting /mnt/c inside the container doesn't help either because the executable still remains unusable for some reason.

**Comment by artm** [comment]
Nevermind, git config was broken in my wsl, It was therefore copied as is to the container and remained broken.

**Comment by Chuxel** [comment]
The latest version of the images and the common utils Dev Container Feature will have the "dirty" check off by default as well.
